src/metabase/task/send_pulses.clj

```diff
@@ -56,16 +60,24 @@
     (catch Throwable e
       (log/errorf e "Error sending Pulse %d to channel ids: %s" pulse-id (str/join ", " channel-ids)))))
 
+(defn- send-trigger-timezone
+  []
+  (or (driver/report-timezone)
+      (qp.timezone/system-timezone-id)
+      "UTC"))
```

Comment:
This feels like it could be DRY'd up with other parts of the codebase outside of pulses. It doesn't feel right to have a notion of the trigger time zone specific to send pulses task.

Also, before we were just using `(driver/report-timezone)` without `(qp.timezone/system-timezone-id)`, can you explain the change?

Reply:
it's the fallback timezone id we used in QP so I think we should apply it for pulse too: https://github.com/metabase/metabase/blob/907da2dd943508e22c9fc2db63fee066df6b270d/src/metabase/query_processor/timezone.clj#L90

It's also used in other tasks like persist refresh : https://github.com/metabase/metabase/blob/send-pulse-trigger-respect-timezone/src/metabase/task/persist_refresh.clj#L286

Reply:
Right, I guess we should DRY this up. Instead of duplicating this logic again, `update-send-pulse-triggers-timezone!`, and the `cron-schedule` function should use the same function. Why not use `results-timezone-id` from the QP code?

Reply:
`results-timezone-id` requires a DB. In case of pulse a dashboard can contains questions from more than 1 DB.

Wdyt if we move this to `metabase.task` or a new `metabase.task.util`?

Reply:
I don't mind where it goes too much. But I'm seeing a couple more places this could be DRYed up too:

https://github.com/metabase/metabase/blob/e6f47956c5a22c8052ae2db201f087cd6dcc4eac/src/metabase/email/messages.clj#L577-L579

https://github.com/metabase/metabase/blob/e6f47956c5a22c8052ae2db201f087cd6dcc4eac/src/metabase/task/index_values.clj#L82-L84